# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1313664968 -7200
#      Thu Aug 18 12:56:08 2011 +0200
# Node ID 1fa83c820402c73f901e4c2e8e359a0959ce5e37
# Parent  f72530c9471adac24372bd9b20f045e8ab0ecd64
Added minimum constraints to subwindows.

diff --git a/gui/src/MainWindow.cpp b/gui/src/MainWindow.cpp
--- a/gui/src/MainWindow.cpp
+++ b/gui/src/MainWindow.cpp
@@ -292,40 +292,43 @@ MainWindow::construct ()
   m_centralMdiArea->addSubWindow (m_octaveTerminalSubWindow, Qt::WindowTitleHint | Qt::WindowMinMaxButtonsHint);
 
   m_octaveTerminalSubWindow->setObjectName ("OctaveTerminalSubWindow");
   m_octaveTerminalSubWindow->setWindowTitle (tr ("Terminal"));
   m_octaveTerminalSubWindow
       ->setWindowIcon (ResourceManager::instance ()->icon (ResourceManager::Terminal));
   m_octaveTerminalSubWindow->setFocusProxy (m_octaveTerminal);
   m_octaveTerminalSubWindow->setStatusTip (tr ("Enter your commands into the Octave terminal."));
+  m_octaveTerminalSubWindow->setMinimumSize (670, 300);
 
   // Documentation subwindow.
   m_documentationWidgetSubWindow = new NonClosableMdiSubWindow (this);
   m_documentationWidgetSubWindow->setWidget (m_documentationWidget);
   m_centralMdiArea->addSubWindow (m_documentationWidgetSubWindow, Qt::WindowTitleHint | Qt::WindowMinMaxButtonsHint);
 
   m_documentationWidgetSubWindow->setObjectName ("DocumentationWidgetSubWindow");
   m_documentationWidgetSubWindow->setWindowTitle (tr ("Documentation"));
   m_documentationWidgetSubWindow
       ->setWindowIcon (ResourceManager::instance ()->icon (ResourceManager::Documentation));
   m_documentationWidgetSubWindow->setFocusProxy (m_documentationWidget);
   m_documentationWidgetSubWindow->setStatusTip (tr ("Browse the Octave documentation for help."));
+  m_documentationWidgetSubWindow->setMinimumSize (670, 300);
 
   // Chat subwindow.
   m_ircWidgetSubWindow = new NonClosableMdiSubWindow (this);
   m_ircWidgetSubWindow->setWidget (m_ircWidget);
   m_centralMdiArea->addSubWindow (m_ircWidgetSubWindow, Qt::WindowTitleHint | Qt::WindowMinMaxButtonsHint);
 
   m_ircWidgetSubWindow->setObjectName ("ChatWidgetSubWindow");
   m_ircWidgetSubWindow->setWindowTitle (tr ("Chat"));
   m_ircWidgetSubWindow
       ->setWindowIcon (ResourceManager::instance ()->icon (ResourceManager::Chat));
   m_ircWidgetSubWindow->setStatusTip(tr ("Instantly chat with other Octave users for help."));
   m_ircWidgetSubWindow->setFocusProxy (m_ircWidget);
+  m_ircWidgetSubWindow->setMinimumSize (670, 300);
   connect (m_ircWidget, SIGNAL (unreadMessages (bool)), this, SLOT (handleUnreadMessages (bool)));
 
   m_lexer = NULL;  // initialise the empty lexer for the edtiors
 
   QMenu *controlMenu = menuBar ()->addMenu (tr ("Octave"));
   QAction *settingsAction = controlMenu->addAction (tr ("Settings"));
   controlMenu->addSeparator ();
   QAction *exitAction = controlMenu->addAction (tr ("Exit"));
diff --git a/gui/src/OctaveTerminal.cpp b/gui/src/OctaveTerminal.cpp
--- a/gui/src/OctaveTerminal.cpp
+++ b/gui/src/OctaveTerminal.cpp
@@ -39,16 +39,17 @@ void
 OctaveTerminal::construct ()
 {
   m_terminalWidget = new QTerminalWidget (0, this);
   m_terminalWidget->setSizePolicy (QSizePolicy::Expanding, QSizePolicy::Expanding);
   setFocusProxy (m_terminalWidget);
 
   QVBoxLayout *layout = new QVBoxLayout ();
   layout->addWidget (m_terminalWidget);
+  layout->setMargin (3);
   setLayout (layout);
 }
 
 void
 OctaveTerminal::openTerminal ()
 {
   int fdm, fds;
   if (openpty (&fdm, &fds, 0, 0, 0) < 0)
